x-pack/plugins/monitoring/public/components/logstash/pipeline_viewer/views/config_viewer/statement_list.js

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { StatementListHeading } from './statement_list_heading';
+import { Statement } from './statement';
+
+export function StatementSection({
+  iconType,
+  headingText,
+  elements,
+  onShowVertexDetails
+}) {
+  return (
+    <div className="cv-statementList">
+      <StatementListHeading
+        iconType={iconType}
+        title={headingText}
+      />
+      <StatementList
+        elements={elements}
+        onShowVertexDetails={onShowVertexDetails}
+      />
+    </div>
+  );
+}
+
+function getCollapsedChildElements(elements, collapsedIds) {
+  const collapsedChildIds = new Set();
+  elements.forEach(({ id, parentId }) => {
+    if (collapsedIds.has(parentId) || collapsedChildIds.has(parentId)) {
+      collapsedChildIds.add(id);
+    }
+  });
+  return collapsedChildIds;
+}
+
+class StatementList extends React.PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      collapsedIds: new Set(),
+      collapsedChildIds: new Set()
+    };
+
+    this.collapse = this.collapse.bind(this);
+    this.expand = this.expand.bind(this);
+    this.renderStatement = this.renderStatement.bind(this);
+    this.elementIsCollapsed = this.elementIsCollapsed.bind(this);
```

Comment:
Nothing wrong with this approach but just an alternative to consider: You can omit these `bind` statements in the constructor and instead define the methods like so:

```
expand = (elementId) => {
 ...
}
```

Reply:
+1, added in https://github.com/elastic/kibana/pull/18597/commits/49a4a7d41b935ea5aed2c08766e5dcc3a8a74fe9